src/service/link-rewriter/link-rewriter-manager.js

```diff
@@ -0,0 +1,231 @@
+import {AmpEvents} from '../../amp-events';
+import {EVENTS, LINK_REWRITER_SERVICE_NAME, PRIORITY_META_TAG_NAME} from './constants';
+import {LinkRewriter} from './link-rewriter';
+import {Services} from '../../services';
+import {registerServiceBuilderForDoc} from '../../service';
+
+/**
+ * LinkRewriterManager works conjointly with LinkRewriter to allow rewriting
+ * links at runtime. E.g: Replacing a link by its affiliate version only if
+ * the link can be monetised. A page can have multiple LinkRewriter running
+ * at the same time.
+ *
+ * LinkRewriterManager class is in charge of:
+ * - Keeping track of all the registered linkRewriters
+ * - Notifying all the linkRewriters when the DOM has changed.
+ * - Managing which LinkRewriter has the priority to replace a link
+ *   (based on configurable priority list).
+ * - Notifying the most relevant LinkRewriter that a click happened
+ *   so the linkRewriter handles the potential replacement.
+ * - Sending a click event to listeners to be able to track the click
+ *   even if the url has not been replaced.
```

Comment:
Perhaps, a short comment repeating that "listening rewriters should not mutate the anchor in their handlers!".